src/stripeTerminal.ts

```diff
@@ -33,28 +33,37 @@ export class StripeTerminal {
 
     const commandString = [cliPath, command, ...args, ...globalCLIFLags].join(' ');
 
-    const terminal = await this.createTerminal();
-    terminal.sendText(commandString);
-    terminal.show();
+    try {
+      const terminal = await this.createTerminal();
+      terminal.sendText(commandString);
+      terminal.show();
+    } catch (e) {
+      vscode.window.showErrorMessage(e.message);
+    }
   }
 
-  private createNewSplitTerminal(): Promise<vscode.Terminal> {
-    return new Promise(async (resolve, reject) => {
-      await vscode.commands.executeCommand('workbench.action.terminal.split');
+  private async createNewSplitTerminal(): Promise<vscode.Terminal | undefined> {
+    const lastTerminal = this.terminals[this.terminals.length - 1];
+    lastTerminal.show();
 
-      vscode.window.onDidChangeActiveTerminal((terminal) => {
-        if (terminal) {
-          resolve(terminal);
-        }
-      });
-    });
+    // Note that this splits off of the user's currently visible terminal. That's why `.show()` is
+    // called above.
+    await vscode.commands.executeCommand<vscode.Terminal>('workbench.action.terminal.split');
+
+    // After `workbench.action.terminal.split`, the activeTerminal becomes the newly split terminal.
+    // Note that there is no API guarantee for this behavior; a prior implementation relied on
+    // different behavior entirely. Because historically this behavior has been in flux, we should
+    // move to an official API for creating split terminals once that lands in vscode.
+    return vscode.window.activeTerminal;
   }
 
   private async createTerminal(): Promise<vscode.Terminal> {
     if (this.terminals.length > 0) {
-      const lastTerminal = this.terminals[this.terminals.length - 1];
-      lastTerminal.show(); // In case it is hidden
       const terminal = await this.createNewSplitTerminal();
+      if (!terminal) {
+        throw new Error('Failed to split Stripe terminal');
```

Comment:
Should we say try again on error? I'm slightly confused what a Stripe terminal is. What do you think?

Reply:
Good question. I don't think we want to retry because to reach this exception, then something very wrong must have happened.

A Stripe terminal is a vscode terminal whose sole purpose is to run `stripe` commands. I'm not sure if there's a better name for it, happy to change it if you have any suggestions.

Reply:
Oops sorry, I misread your comment - yeah we can say try again on error! And then we won't have to mention anything about a stripe terminal.